src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.ArrayList;
+
+/**
+ * Interpret and process user inputs.
+ */
+public class Parser {
+    protected TaskList tasks;
+    protected Storage storage;
+    protected Ui ui;
+
+    /**
+     * Saves tasks, storage and ui class.
+     *
+     * @param tasks The TaskList class.
+     * @param storage The Storage class.
+     * @param ui The Ui class.
+     */
+    public Parser(TaskList tasks, Storage storage, Ui ui) {
+        this.tasks = tasks;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /**
+     * Processes the user input.
+     *
+     * @param userInput The commands issued to duke by the user.
+     * @return Returns false when user wants to terminate the program.
+     */
+    public boolean parse(String userInput) {
+        if (userInput.equals("bye")) {
```

Comment:
You can consider using the method equalsIgnoreCase() to make your bot commands more flexible.